letsencrypt/client/reverter.py

```diff
@@ -0,0 +1,434 @@
+                self._recover_checkpoint(self.direc['temp'])
+            except errors.LetsEncryptReverterError:
+                # We have a partial or incomplete recovery
+                logging.fatal("Incomplete or failed recovery for %s",
+                              self.direc['temp'])
+                raise errors.LetsEncryptReverterError(
+                    "Unable to revert temporary config")
+
+    def rollback_checkpoints(self, rollback=1):
+        """Revert 'rollback' number of configuration checkpoints.
+
+        :param int rollback: Number of checkpoints to reverse. A str num will be
+           cast to an integer. So '2' is also acceptable.
+
+        :raises :class:`letsencrypt.client.errors.LetsEncryptReverterError`: If
+            there is a problem with the input or if the function is unable to
+            correctly revert the configuration checkpoints.
+
+        """
+        try:
+            rollback = int(rollback)
+        except ValueError:
+            logging.error("Rollback argument must be a positive integer")
+            raise errors.LetsEncryptReverterError("Invalid Input")
+        # Sanity check input
+        if rollback < 0:
+            logging.error("Rollback argument must be a positive integer")
+            raise errors.LetsEncryptReverterError("Invalid Input")
+
+        backups = os.listdir(self.direc['backup'])
+        backups.sort()
+
+        if len(backups) < rollback:
+            logging.warning("Unable to rollback %d checkpoints, only %d exist",
+                            rollback, len(backups))
+
+        while rollback > 0 and backups:
+            cp_dir = os.path.join(self.direc['backup'], backups.pop())
+            try:
+                self._recover_checkpoint(cp_dir)
+            except errors.LetsEncryptReverterError:
+                logging.fatal("Failed to load checkpoint during rollback")
+                raise errors.LetsEncryptReverterError(
+                    "Unable to load checkpoint during rollback")
+            rollback -= 1
+
+    def view_config_changes(self):
+        """Displays all saved checkpoints.
+
+        All checkpoints are printed to the console.
+
+        .. todo:: Decide on a policy for error handling, OSError IOError...
+
+        """
+        backups = os.listdir(self.direc['backup'])
+        backups.sort(reverse=True)
+
+        if not backups:
+            logging.info("The Let's Encrypt client has not saved any backups "
+                         "of your configuration")
+            return
+        # Make sure there isn't anything unexpected in the backup folder
+        # There should only be timestamped (float) directories
+        try:
+            for bkup in backups:
+                float(bkup)
+        except ValueError:
+            raise errors.LetsEncryptReverterError(
+                "Invalid directories in {0}".format(self.direc['backup']))
+
+        for bkup in backups:
+            print time.ctime(float(bkup))
```

Comment:
As I feared... it wasn't a simple fix.  I had to modify the display class functions...and the API... since I was editing the display API... I went ahead and documented all of the functions while I was at it...just to keep everything consistent.  The class still needs some work... but at least the functions are documented and the notifications work.

view_config_changes now correctly uses the display output of choice.  I am not 100% satisfied with the FileDisplay generic_notification especially... but it is a strict improvement on what is there.

I did do a documentation check to make sure that SPHINX ran without warnings.